storage/localstore/localstore.go

```diff
@@ -336,6 +342,50 @@ func New(path string, baseKey []byte, o *Options) (db *DB, err error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Create a index structure for storing pinned chunks and their pin counts
+	db.pinIndex, err = db.shed.NewIndex("Hash->pinCounter", shed.IndexFuncs{
+		EncodeKey: func(fields shed.Item) (key []byte, err error) {
+			return fields.Address, nil
+		},
+		DecodeKey: func(key []byte) (e shed.Item, err error) {
+			e.Address = key
+			return e, nil
+		},
+		EncodeValue: func(fields shed.Item) (value []byte, err error) {
+			b := make([]byte, 8)
+			binary.BigEndian.PutUint64(b[:8], fields.PinCounter)
```

Comment:
you're using a `PutUint64` function to marshal and unmarshal a `uint8`. Why not use `binary.PutUvarint`? also please note the possible overflow in this case

Reply:
I think we should make pin counter to uint64 itself.. instead of uint8.

Reply:
Done